### ISSUE TITLE
the gem does not seem to work with gh pages

### ISSUE THREAD

**Issue by KaMuench** [NiklasEi/jekyll_custom_permalink#4]
It seems that gh pages is uable to use the dependency in the Gemfile. In the deployment of my gh pages repository it says:

-> github-pages can't satisfy your Gemfile's dependencies.

Since I added the gem , git hub does not deploy the site anymore.

**Comment by NiklasEi** [comment]
Is there any other relevant output in the logs?

**Comment by KaMuench** [comment]
The build with jekyll section contains following log:

Run actions/jekyll-build-pages@v1
/usr/bin/docker run --name ghcrioactionsjekyllbuildpagesv107_1e06ca --label ed866e --workdir /github/workspace --rm -e "INPUT_SOURCE" -e "INPUT_DESTINATION" -e "INPUT_FUTURE" -e "INPUT_BUILD_REVISION" -e "INPUT_VERBOSE" -e "INPUT_TOKEN" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e "ACTIONS_ID_TOKEN_REQUEST_URL" -e "ACTIONS_ID_TOKEN_REQUEST_TOKEN" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/blog/blog":"/github/workspace" ghcr.io/actions/jekyll-build-pages:v1.0.7
Warning:  github-pages can't satisfy your Gemfile's dependencies.
  Logging at level: debug
Configuration file: /github/workspace/./_config.yml
             Theme: minima
      Theme source: /usr/local/bundle/gems/minima-2.5.1
      GitHub Pages: github-pages v228
      GitHub Pages: jekyll v3.9.3
             Theme: minima
      Theme source: /usr/local/bundle/gems/minima-2.5.1
         Requiring: jekyll-feed
         Requiring: jekyll-seo-tag
         Requiring: jekyll-feed
         Requiring: jekyll-coffeescript
         Requiring: jekyll-commonmark-ghpages
         Requiring: jekyll-gist
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
  Liquid Exception: The URL template doesn't have subtitle keys. Check your permalink template! in /_layouts/default.html
/usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/url.rb:100:in `block in generate_url_from_drop': The URL template doesn't have subtitle keys. Check your permalink template! (NoMethodError)
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/url.rb:95:in `gsub'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/url.rb:95:in `generate_url_from_drop'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/url.rb:62:in `generate_url'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/url.rb:53:in `generated_url'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/url.rb:39:in `to_s'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/document.rb:211:in `url'
	from /usr/local/lib/ruby/2.7.0/forwardable.rb:235:in `url'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/drops/drop.rb:52:in `public_send'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/drops/drop.rb:52:in `[]'
	from /usr/local/bundle/gems/jekyll-seo-tag-2.8.0/lib/jekyll-seo-tag/drop.rb:171:in `canonical_url'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/drops/drop.rb:52:in `public_send'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/drops/drop.rb:52:in `[]'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/context.rb:197:in `lookup_and_evaluate'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/variable_lookup.rb:48:in `block in evaluate'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/variable_lookup.rb:38:in `each_index'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/variable_lookup.rb:38:in `evaluate'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/context.rb:159:in `evaluate'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/variable.rb:82:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:103:in `render_node_to_output'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:80:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:46:in `block (2 levels) in render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:44:in `each'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:44:in `block in render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/context.rb:123:in `stack'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:43:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:103:in `render_node_to_output'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:82:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:206:in `block in render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:240:in `with_profiling'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:205:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:218:in `render!'
	from /usr/local/bundle/gems/jekyll-seo-tag-2.8.0/lib/jekyll-seo-tag.rb:36:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:103:in `render_node_to_output'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:91:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:206:in `block in render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:240:in `with_profiling'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:205:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:218:in `render!'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/liquid_renderer/file.rb:30:in `block (2 levels) in render!'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/liquid_renderer/file.rb:42:in `measure_bytes'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/liquid_renderer/file.rb:29:in `block in render!'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/liquid_renderer/file.rb:49:in `measure_time'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/liquid_renderer/file.rb:28:in `render!'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/tags/include.rb:140:in `block in render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/context.rb:123:in `stack'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/tags/include.rb:137:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:103:in `render_node_to_output'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:91:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:206:in `block in render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:240:in `with_profiling'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:205:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:218:in `render!'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/liquid_renderer/file.rb:30:in `block (2 levels) in render!'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/liquid_renderer/file.rb:42:in `measure_bytes'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/liquid_renderer/file.rb:29:in `block in render!'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/liquid_renderer/file.rb:49:in `measure_time'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/liquid_renderer/file.rb:28:in `render!'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/renderer.rb:126:in `render_liquid'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/renderer.rb:195:in `render_layout'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/renderer.rb:158:in `place_in_layouts'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/renderer.rb:88:in `render_document'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/renderer.rb:62:in `run'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/site.rb:479:in `render_regenerated'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/site.rb:464:in `block (2 levels) in render_docs'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/site.rb:463:in `each'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/site.rb:463:in `block in render_docs'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/site.rb:462:in `each_value'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/site.rb:462:in `render_docs'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/site.rb:191:in `render'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/site.rb:71:in `process'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/command.rb:28:in `process_site'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/commands/build.rb:65:in `build'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/commands/build.rb:36:in `process'
	from /usr/local/bundle/gems/github-pages-228/bin/github-pages:70:in `block (3 levels) in <top (required)>'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
	from /usr/local/bundle/gems/github-pages-228/bin/github-pages:6:in `<top (required)>'
	from /usr/local/bundle/bin/github-pages:23:in `load'
	from /usr/local/bundle/bin/github-pages:23:in `<main>'
         Requiring: jekyll-github-metadata
         Requiring: jekyll-paginate
         Requiring: jekyll-relative-links
         Requiring: jekyll-optional-front-matter
         Requiring: jekyll-readme-index
         Requiring: jekyll-default-layout
         Requiring: jekyll-titles-from-headings
   GitHub Metadata: Initializing...
            Source: /github/workspace/.
       Destination: /github/workspace/./_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
       EntryFilter: excluded /Gemfile
       EntryFilter: excluded /Gemfile.lock
           Reading: _posts/2023-05-10-post-test.md
           Reading: _posts/2023-04-27-welcome-to-jekyll.markdown
           Reading: _posts/2023-05-10-second-post-test.md
           Reading: _projects/project_test.md
        Generating: JekyllOptionalFrontMatter::Generator finished in 2.89e-05 seconds.
        Generating: JekyllReadmeIndex::Generator finished in 4.0401e-05 seconds.
       Jekyll Feed: Generating feed for posts
        Generating: JekyllFeed::Generator finished in 0.000636311 seconds.
        Generating: Jekyll::Paginate::Pagination finished in 4.5e-06 seconds.
        Generating: JekyllRelativeLinks::Generator finished in 0.000220404 seconds.
        Generating: JekyllDefaultLayout::Generator finished in 0.001813831 seconds.
        Generating: JekyllTitlesFromHeadings::Generator finished in 1.8e-05 seconds.
         Rendering: _posts/2023-04-27-welcome-to-jekyll.markdown
  Pre-Render Hooks: _posts/2023-04-27-welcome-to-jekyll.markdown
  Rendering Liquid: _posts/2023-04-27-welcome-to-jekyll.markdown
         Requiring: rouge
         Requiring: kramdown-parser-gfm
  Rendering Markup: _posts/2023-04-27-welcome-to-jekyll.markdown
         Requiring: kramdown-parser-gfm
  Rendering Layout: _posts/2023-04-27-welcome-to-jekyll.markdown
     Layout source: site
         Rendering: _posts/2023-05-10-post-test.md
  Pre-Render Hooks: _posts/2023-05-10-post-test.md
  Rendering Markup: _posts/2023-05-10-post-test.md
  Rendering Layout: _posts/2023-05-10-post-test.md
     Layout source: site
         Rendering: _posts/2023-05-10-second-post-test.md
  Pre-Render Hooks: _posts/2023-05-10-second-post-test.md
  Rendering Markup: _posts/2023-05-10-second-post-test.md
  Rendering Layout: _posts/2023-05-10-second-post-test.md
     Layout source: site
         Rendering: _projects/project_test.md
  Pre-Render Hooks: _projects/project_test.md
  Rendering Markup: _projects/project_test.md
  Rendering Layout: _projects/project_test.md
     Layout source: site
**github-pages 228 | Error:  The URL template doesn't have subtitle keys. Check your permalink template!**

---

It seems that that git hub pages can not use the gem and therefore does not find the `subtitle` placeholder, which I defined in my [_config.yml](https://github.com/KaMuench/blog/blob/gh-pages/_config.yml) file.

**Comment by KaMuench** [comment]
I am not sure what exactly fixed it, but I found a solution and the build now succeeds. I have read in some Stack Overflow posts that git hub builds the jekyll project with the `--safe' parameter, which prevents the use of custom plugins. Since I know little to nothing about Ruby or Jekyll, I'm not sure if this gem could be a custom plugin or something. Anyway, they suggested building the Jekyll project with a custom git hub workflow, and that did the trick. I added a custom workflow and pushed some changes to the branch and the build was successful. 

Since this solved my problem, I will close this issue. But thanks anyway!
